components/stepline.tsx

```diff
@@ -0,0 +1,152 @@
+                hoveredIndex?: number | null;
+              }>
+          >(child)
+        )
+          return null;
+        const variant = child.props.variant;
+        const onClick = child.props.onClick;
+
+        return (
+          <div
+            key={index}
+            onClick={() => {
+              if (reversable && variant) {
+                onClick();
+              }
+            }}
+            onMouseEnter={() => {
+              setHoveredIndex(index + 1);
+            }}
+            onMouseLeave={() => {
+              setHoveredIndex(null);
+            }}
+            className={cn(
+              "flex flex-row gap-2 relative",
+              reversable && variant && "cursor-pointer"
+            )}
+          >
+            <div className="flex flex-row w-full justify-between">
+              {cloneElement(child, {
+                id: index + 1,
+                hoveredIndex: hoveredIndex,
+              })}
+            </div>
+            <div className="flex flex-row relative items-center w-full">
+              {index !== childArray.length - 1 && (
+                <div
+                  className={cn(
+                    "flex-1 h-px min-w-[44px] bg-gray-400",
+                    variant === "error" && "bg-[#b60000]",
+                    (variant === "completed" || variant === "current") &&
+                      "bg-[#00b62e]"
+                  )}
+                />
+              )}
+            </div>
+          </div>
+        );
+      })}
+    </div>
+  );
+}
+
+function SteplineItem({
+  subtitle,
+  title,
+  variant = "todo",
+  className,
+  id,
+  hoveredIndex,
+}: SteplineItemProps &
+  Partial<{
+    hoveredIndex?: number | null;
+  }>) {
+  const steplineItemId = `stepline-item-${id}`;
+
+  return (
+    <div
+      id={steplineItemId}
+      className={cn(
+        "flex flex-row w-full gap-2 items-center whitespace-nowrap",
+        className
+      )}
+    >
+      <div className="flex relative">
+        <div
+          className={cn(
+            "text-white absolute flex items-center justify-center min-w-[30px] min-h-[30px] max-w-[30px] max-h-[30px] transform duration-200 bg-gray-600 rounded-full -translate-y-1/2 top-1/2",
+            hoveredIndex === id && "scale-[130%] bg-gray-400",
+            variant === "error" && "bg-[#8f0751]",
+            (variant === "completed" || variant === "current") && "bg-[#2fe620]"
+          )}
+        />
+        <div
+          className={cn(
+            "text-white flex relative items-center justify-center min-w-[30px] min-h-[30px] max-w-[30px] max-h-[30px] bg-gray-600 rounded-full",
+            variant === "error" && "bg-[#b60000]",
+            (variant === "completed" || variant === "current") && "bg-[#00b62e]"
+          )}
+        >
+          {id}
+        </div>
+      </div>
+      {(title || subtitle) && (
+        <div className="flex flex-col">
+          {title && <h2 className="font-medium text-sm">{title}</h2>}
```

Comment:
let's use `<span>` instead of `h2` ... it's ok, let's use `h2` only for actual page section :)